ompi/request/request.c

```diff
@@ -211,3 +248,19 @@ int ompi_request_persistent_noop_create(ompi_request_t** request)
     *request = req;
     return OMPI_SUCCESS;
 }
+
+#if MPI_VERSION >= 4
+int mca_pml_cancel_send_callback(struct ompi_request_t *request, int flag)
+{
+    static opal_atomic_int32_t send_deprecate_count = 0;
+    int32_t val;
+
+    val = opal_atomic_add_fetch_32(&send_deprecate_count, 1);
+    if (((1 == ompi_pml_base_deprecation_warn_level) && (1 == val)) ||
+               (2 == ompi_pml_base_deprecation_warn_level)) {
+        opal_output(0, "Calling MPI_Cancel for a request created by a non-blocking send is deprecated.");
+        send_deprecate_count++;
```

Comment:
Why the non-atomic increment here?

Reply:
Actually, I don't think this line is needed given the atomic increment a few lines above.